### PR TITLE
Created core ents and implemented createJob mutation

### PR DESCRIPTION
**Pull request by KCarretto**
#### What type of PR is this?

/kind feature

/kind api-change

#### What this PR does / why we need it:

Updates our ents, adds appropriate fields, uses mixins and hooks for optimal code reuse and bug avoidance, and implements the `createJob` mutation (with tests!).

#### Which issue(s) this PR fixes:

Maybe I should make issues first

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
 	// SPEEEEEEED on MacOS & Windows. Note that your local clone of realm will not be updated.
 	"workspaceMount": "source=realm,target=/workspaces/realm,type=volume",
 	"workspaceFolder": "/workspaces/realm",
-	"postCreateCommand": "cd /workspaces && git clone ssh://git@github.com/kcarretto/realm", 
+	"postCreateCommand": "cd /workspaces && git clone https://github.com/KCarretto/realm.git",
```
Comment (hulto) [inline .devcontainer/devcontainer.json:43]: These changes should probably be a separate PR.

File: tavern/ent/schema/job.go [hunk 2]
```diff
@@ -29,19 +29,37 @@ func (Job) Fields() []ent.Field {
...
+			Comment("Tome that this job will be executing").
 			Required().
 			Unique(),
-		edge.To("tasks", Task.Type),
+		edge.To("bundle", File.Type).
+			Annotations(
+				entgql.Skip(entgql.SkipMutationCreateInput),
+			).
+			Comment("Bundle file that the executing tome depends on (if any)").
+			Unique(),
```
Comment (hulto) [inline tavern/ent/schema/job.go:41]: Is this the file resources used inside the tome?

File: tavern/ent/schema/task.go [hunk 3]
```diff
@@ -15,13 +15,30 @@ type Task struct {
 // Fields of the Task.
 func (Task) Fields() []ent.Field {
 	return []ent.Field{
-		field.String("name").
```
Comment (hulto) [inline tavern/ent/schema/task.go:18]: No task names? Are we only using the unique ID to reference tasks?

File: tavern/graphql/query_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	graph := enttest.Open(t, "sqlite3", "file:ent?mode=memory&cache=shared&_fk=1")
+	defer graph.Close()
+
+	// Initialize sample data
+	testUser := graph.User.Create().
+		SetName("bobdylan").
+		SetIsActivated(false).
+		SetIsAdmin(true).
+		SetOAuthID("likearollingstone").
+		SetPhotoURL("https://upload.wikimedia.org/wikipedia/commons/0/02/Bob_Dylan_-_Azkena_Rock_Festival_2010_2.jpg").
```
Comment (hulto) [inline tavern/graphql/query_test.go:30]: Any chance we can use a file we manage maybe in the realm repo?

**Review by hulto — CHANGES_REQUESTED** [review]
Few small changes:
- Move .devcontainer changes to a seperate PR.
- Change test to use managed resources.

File: tavern/ent/schema/task.go (hunk 3 above)
Comment (KCarretto) [reply tavern/ent/schema/task.go:18]: Jobs have the name, and tasks have an edge to the job

File: tavern/ent/schema/job.go (hunk 2 above)
Comment (KCarretto) [reply tavern/ent/schema/job.go:41]: Yessir o7

File: tavern/graphql/query_test.go (hunk 4 above)
Comment (hulto) [reply tavern/graphql/query_test.go:30]: Doesn't do anything with URL can be garbage string.

Comment (KCarretto) [reply tavern/graphql/query_test.go:30]: File is not actually used at all, so this change is unnecessary
